cmd/controller/main.go

```diff
@@ -37,7 +35,6 @@ func main() {
 		op.AMIProvider,
 		op.SecurityGroupProvider,
 	)
-	lo.Must0(op.AddHealthzCheck("cloud-provider", awsCloudProvider.LivenessProbe))
```

Comment:
We've generally been shying away from adding liveness probes to the system just for mutex locks. If there is a deadlock in the system, we've taken the stance that rather than masking it and being overly defensive about it, we should let the situation happen and then fix the root cause.

Restarting on a deadlock just masks the problem.

Reply:
This is true of our SecurityGroupProvider and SubnetProviders where we have mutexes but liveness probes are removed.